modules/data.land/inst/doanload_NEON_multi_site.R

```diff
@@ -0,0 +1,111 @@
+    end_date <- start_date %m+% months(1)
+    
+    #download
+    neonstore::neon_download("DP1.10098.001", dir = store_dir, table = NA, site = tempNEON$siteCode, start_date = start_date, end_date = end_date, type = "basic",api = "https://data.neonscience.org/api/v0")
+    
+    #read
+    apparentindividual <- neonstore::neon_read(table = "apparentindividual", product = "DP1.10098.001", site = tempNEON$siteCode, start_date = start_date, end_date = end_date, dir = store_dir)
+    
+    #re-download if data is problematic
+    while (is.null(apparentindividual)) {
+      print(paste0("current year: ", closeYear, " is problematic. Try next year"))
+      closeYear <- closeYear + 1
+      target_date <- lubridate::date(paste0(closeYear,"-07-15"))
+      daysDiff <- abs(lubridate::days(lubridate::date(NEON_dates)-lubridate::date(target_date))@day)
+      
+      #search within 1 month
+      start_date <- ymd(as.Date(NEON_dates[which(daysDiff == min(daysDiff))]))
+      end_date <- start_date %m+% months(1)
+      
+      #download
+      neonstore::neon_download("DP1.10098.001", dir = store_dir, table = NA, site = tempNEON$siteCode, start_date = start_date, end_date = end_date, type = "basic",api = "https://data.neonscience.org/api/v0")
+      apparentindividual <- neonstore::neon_read(table = "apparentindividual", product = "DP1.10098.001", site = tempNEON$siteCode, start_date = start_date, end_date = end_date, dir = store_dir)
+    }
+    #
+    print(paste0("find good year: ", closeYear, ", for site: ", tempNEON$siteCode))
+    log_txt <- c(log_txt, paste0("find good year: ", closeYear, ", for site: ", tempNEON$siteCode))
+    
+    mappingandtagging <- neonstore::neon_read(table = "mappingandtagging", product = "DP1.10098.001", site = tempNEON$siteCode, start_date = start_date, end_date = end_date, dir = store_dir)
+    joined.veg <- dplyr::left_join(mappingandtagging, apparentindividual, by = "individualID")
+    
+    #Filter joined.veg for required information: DBH, tree height, and species
+    filter.veg <- dplyr::select(joined.veg, .data$siteID.x, .data$plotID.x, .data$subplotID, .data$taxonID, .data$scientificName, .data$taxonRank, .data$date.y, .data$stemDiameter, .data$height)
+    
+    #Filter for most recent record
+    filter.date <- dplyr::filter(filter.veg, .data$date.y >= start_date)
+    
+    #Create year column
+    filter.date$year <- format(as.Date(filter.date$date.y, format="%d/%m/%Y"),"%Y")
+    
+    #Rename NEON column names to match pecan functions
+    colnames(filter.date) <- c("site_name", "plot", "Subplot", "species_USDA_symbol", "species", "taxonRank", "date", "DBH", "height", "year")
+    
+    #Create veg_info object as a list
+    veg_info <- list()
+    
+    #Set filter.date as veg_info[[2]]
+    veg_info[[2]] <- filter.date
+    
+    #Set plot size as veg_info[[1]]
+    veg_info[[1]] <- list(area = 400)
+    
+    #set site ID
+    veg_info[[3]] <- settings[[i]]$run$site$id
```

Comment:
Is downstream code expecting this? Would it be better to do something line names(VEG_INFO)[i] <- settings[[i]]$run$site$id

Reply:
Higher level -- given that there's already a function to download NEON for a single site, why is this function duplicating that functionality rather than just calling that function within the for loop?